azure/Guidelines.md

```diff
@@ -0,0 +1,864 @@
+Services, and the clients that access them, may be written in multiple languages. To ensure interoperability, JSON establishes the "lowest common denominator" type system, which is always sent over the wire as UTF-8 bytes. This system is very simple and consists of three types:
+
+ Type | Description
+ ---- | -----------
+ Boolean | true/false (always lowercase)
+ Number  | Signed floating point (IEEE-754 binary64; int range: -2<sup>53</sup>+1 to +2<sup>53</sup>-1)
+ String  | Used for everything else
+
+> :white_check_mark: **DO** use integers within the acceptable range of JSON number.
+> 
+> :white_check_mark: **DO** establish a well-defined contract for the format of strings. For example, determine maximum length, legal characters, case-(in)sensitive comparisons, etc. Where possible, use standard formats, e.g. RFC3339 for date/time.
+> 
+> :white_check_mark: **DO** use strings formats that are well-known and easily parsable/formattable by many programming languages, e.g. RFC3339 for date/time.
+> 
+> :white_check_mark: **DO** ensure that information exchanged between your service and any client is "round-trippable" across multiple programming languages.
+> 
+> :white_check_mark: **DO** use [RFC3339] for date/time.
+> 
+> :white_check_mark: **DO** use [RFC4122] for UUIDs.
+
+> :heavy_check_mark: **YOU MAY** use JSON objects to group sub-fields together.
+> 
+> :heavy_check_mark: **YOU MAY** use JSON arrays if maintaining an order of values is required. Avoid arrays in other situations since arrays can be difficult and inefficient to work with, especially with ```JSON Merge Patch``` where the entire array needs to be read prior to any operation being applied to it.
+
+> :ballot_box_with_check: **YOU SHOULD** use JSON objects instead of arrays whenever possible.
+
+#### Enums & SDKs (Client libraries)
+It is common for strings to have an explicit set of values. These are often reflected in the OpenAPI definition as enumerations. These are extremely useful for developer tooling, e.g. code completion, and client library generation.
+
+However, it is not uncommon for the set of values to grow over the life of a service. For this reason, Microsoft's tooling uses the concept of an "extensible enum," which indicates that the set of values should be treated as only a *partial* list.
+This indicates to client libraries and customers that values of the enumeration field should be effectively treated as strings and that undocumented value may returned in the future. This enables the set of values to grow over time while ensuring stability in client libraries and customer code.
+
+> :ballot_box_with_check: **YOU SHOULD** use extensible enumerations unless you are positive that the symbol set will NEVER change over time.
+> 
+> :white_check_mark: **DO** document to customers that new values may appear in the future so that customers write their code today expecting these new values tomorrow.
+ 
+> :no_entry: **DO NOT** send "enum integers" over the wire.
```

Comment:
It might be good to add justification for such strong guideline.

Reply:
+1. Will add some clarity.

Reply:
I am going to remove this guidance and opened #271 to document the problem and drive resolution.